drud/ddev
# Better error handling when opening/writing config files, fixes  #1177

## The Problem/Issue/Bug:
#1177: Naive error checking caused nil pointer dereference panics when the error returned was indicating something other than the file not existing.
 
## How this PR Solves The Problem:
This change prevents the panic by expanding the error case to _any_ error value, not just those indicating the file doesn't exist. In the specific case encountered in #1177, this means that if the file exists, but ddev doesn't have permissions to read or edit it, we won't try to.

## Manual Testing Instructions:
- `mkdir -p panic/sites/default`
- `cd panic`
- `sudo chown root sites`
- `sudo chmod 000 sites`
- `ddev config --project-name panic --docroot . --project-type drupal8`
- Ensure that ddev warns that it doesn't have permissions to write config files and .gitignore, but no panic is encountererd

## Automated Testing Overview:


## Related Issue Link(s):
#1177
